base/stream.jl

```diff
@@ -552,6 +552,22 @@ mutable struct Pipe <: AbstractPipe
     in::PipeEndpoint # writable
     out::PipeEndpoint # readable
 end
+
+"""
+Construct an uninitialized Pipe object.
+
+The appropriate end of the pipe will be automatically initialized if
+the object is used in process spawning. This can be useful to easily
+obtain references in process pipelines, e.g.:
+
+```
+julia> err = Pipe()
+
+# After this `p` will be initialized and you may read `foo`'s
+# stderr from the `err` pipe.
+julia> spawn(pipeline(pipeline(`foo`, stderr=err), `cat`)
```

Comment:
The command is missing a closing )

Reply:
Personally I think we could also include an example with this syntax:

> spawn(`cat`, stdin, stdout, stderr)

Reply:
The 4-argument `spawn()` appears to be [undocumented](https://docs.julialang.org/en/latest/stdlib/base/#Base.spawn) though.